src/parser/Parser.ts

```diff
@@ -2446,6 +2450,18 @@ export class Parser {
             // just get whatever's next
             typeToken = this.advance();
         }
+
+        if (typeToken && this.options.mode === ParseMode.BrighterScript) {
+            // Check if it is an array - that is, if it has `[]` after the type
+            // eg. `string[]` or `SomeKlass[]`
+            if (this.check(TokenKind.LeftSquareBracket)) {
+                this.advance();
+                if (this.check(TokenKind.RightSquareBracket)) {
+                    const rightBracket = this.advance();
+                    typeToken = createToken(TokenKind.Identifier, typeToken.text + '[]', util.getRange(typeToken, rightBracket));
```

Comment:
I think this would work much better long-term if we added a `TypeExpression` instead, which would support all sorts of this type of pattern. 

We don't need to do that in this PR though, we can tweak it in the future.